src/ADTs.jl

```diff
@@ -0,0 +1,215 @@
+module ADTs
+export Value, Kwarg, Args, Statement, ACSetSpec,
+  construct, acsetspec, generate_expr,
+  to_string, to_dict, label2index
+
+using MLStyle
+using ..ACSetInterface
+# using Catlab
+# using Catlab.CategoricalAlgebra
+
+@data Args{T} begin
+  Value(T)
+  Kwarg(Symbol, Value{T})
+end
+
+@as_record struct Statement
+  table::Symbol
+  element::Vector{Args}
+end
+
+@as_record struct ACSetSpec
+  acstype::Union{Symbol,Expr}
+  body::Vector{Statement}
+end
+
+"""    to_string(s::ACSetSpec)
+
+generates a human readable string of the `ACSetSpec` (or any sub-term).
+"""
+function to_string(s)
```

Comment:
Why do we need this? Can't we just print the result of generate_expr? Also, can we overload some existing function, like `show` maybe, instead of making a new function with a very generic name.

Reply:
The idiomatic thing would be to define `show(io::IO, ...)` and then call `sprint(show, s)` when you want a string.

Reply:
When you recursively call show on strings:

```
sprint(show, gspec) = "\"LabeledGraph{Symbol} begin \\n\\\"  V(\\\\\\\"label=\\\\\\\":a)\\\"\\n\\\"  V(\\\\\\\"label=\\\\\\\":b)\\\"\\n\\\"  V(\\\\\\\"label=\\\\\\\":c)\\\"\\n\\\"  E(1,3)\\\"\\n\\\"  E(2,3)\\\"\\n end\""
```

Julia keeps escaping the backslashes.

Reply:
fixed it by using print in the leaves of the recursion.